app/[locale]/faculty-and-staff/client-utils.tsx

```diff
@@ -0,0 +1,177 @@
+            className="font-semibold text-shade-dark"
+            variant={'link'}
+            onClick={() =>
+              window.history.replaceState(null, '', `?department=${urlName}`)
+            }
+          >
+            {name}
+          </Button>
+        </li>
+      ))}
+    </ol>
+  );
+};
+
+export const FilteredFacultyList = ({
+  departments,
+  departmentHeads,
+  deptartmentHeadText,
+  faculty,
+  locale,
+}: {
+  departments: Map<string, number>;
+  departmentHeads: Set<number>;
+  deptartmentHeadText: string;
+  faculty: {
+    employeeId: string;
+    id: number;
+    person: {
+      email: string;
+      name: string;
+      telephone: string;
+    };
+    departmentId: number;
+    designation:
+      | 'Assistant Professor Grade-I'
+      | 'Assistant Professor Grade-II'
+      | 'Associate Professor'
+      | 'Professor';
+  }[];
+  locale: string;
+}) => {
+  const searchParams = useSearchParams();
+  const query = searchParams.get('query') ?? '';
+  const currentDepartmentId = departments.get(
+    searchParams.get('department') ?? ''
+  );
+  return faculty
+    .filter(
+      ({ person, departmentId }) =>
+        person.name.toLowerCase().includes(query.toLowerCase()) &&
+        (!currentDepartmentId || departmentId === currentDepartmentId)
+    )
+    .map((faculty, index) => (
+      <li
+        className="rounded border border-primary-700 bg-neutral-50 hover:drop-shadow-md"
+        key={index}
+      >
+        <Link
+          className="flex gap-4 p-2 sm:p-3 md:p-4"
+          href={`/${locale}/faculty-and-staff/${faculty.employeeId}`}
+        >
+          <Image
+            alt={faculty.person.name}
+            className="size-32 rounded lg:size-36 xl:size-40 2xl:size-44"
+            height={0}
+            src={`persons/${faculty.id}/image.png`}
+            width={0}
+          />
+          <main>
+            <header className="mb-1 sm:mb-2 md:mb-3 lg:mb-4">
+              <h4 className="mb-0">{faculty.person.name}</h4>
+              <p>
+                {faculty.designation}
+                {departmentHeads.has(faculty.id) && ` (${deptartmentHeadText})`}
+              </p>
+            </header>
+
+            <ul>
+              <li className="flex items-center gap-2">
+                <MdEmail className="fill-primary-700" />
+                {faculty.person.email}
+              </li>
+              <li className="flex items-center gap-2">
+                <FaPhone className="fill-primary-700" />
+                {faculty.person.telephone}
+              </li>
+            </ul>
+          </main>
+        </Link>
+      </li>
+    ));
+};
+
+export const SearchInput = ({
+  defaultValue,
+  placeholder,
+}: {
+  defaultValue?: string;
+  placeholder: string;
+}) => {
+  const debounceCallback = useDebounceCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      window.history.replaceState(null, '', `?query=${event.target.value}`);
```

Comment:
Huh, how does this work? This won't trigger a re-fetch?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState